### ISSUE TITLE
Discount fixed amount whole cart applied mutiple time when customer use Check Out with Multiple Addresses

### ISSUE THREAD

**Issue by hoangnm89** [magento/magento2#25834]
### Preconditions (*)

1. Fresh Magento 2.3.3

### Steps to reproduce (*)
0. Create cart rule, no condition, action: Fixed amount discount for whole cart. Discount amount $10
1. Create account, create 2 addresses: address 1, address 2
2. Add 2 SKUs to cart: SKU1, SKU2
3. Check Out with Multiple Addresses
4. SKU1 ship to address 1, SKU2 ship to address 2
5. Place order
6. Magento auto split order into 2 orders.

### Expected result (*)
1. First Order get discount $10.
2. Second Order should not get discount.

### Actual result (*)

1. Both order get $10 discount, Total discount is $20

=============
According magento comment in code:
https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/SalesRule/Model/Rule/Action/Discount/CartFixed.php#L77

```
/**
 * prevent applying whole cart discount for every shipping order, but only for first order
 */
if ($quote->getIsMultiShipping()) {
    $usedForAddressId = $this->getCartFixedRuleUsedForAddress($rule->getId());
    if ($usedForAddressId && $usedForAddressId != $address->getId()) {
        return $discountData;
    } else {
        $this->setCartFixedRuleUsedForAddress($rule->getId(), $address->getId());
    }
}
```

=== UPDATE: Dec 04 ===
Please cover this case as well
SKU 1: 100$
SKU 2: 200$
Rule: Fix amount 150$ for whole cart(CartFixed)
i add to cart:
SKU 1 - qty 1
SKU 2 - qty 1
then checkout:
SKU 1 - address 1
SKU 2 - address 2

### Expectation:
after checkout:
Sale Order 1: discount amount 100$
Sale Order 2: discount amount: 50$

**Comment by cafefin** [comment]
@magento give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @cafefin. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @cafefin, here is your [Magento instance](https://i-25834-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-25834-2-3-develop.instances.magento-community.engineering/admin](https://i-25834-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by hoangnm89** [comment]
i can replicate the issue

![image](https://user-images.githubusercontent.com/655821/69859596-092da680-12c7-11ea-8108-0dfef3ddb3c4.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @krishprakash 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29276` were created

**Issue Available**: @krishprakash, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by hoangnm89** [comment]
hi @magento-engcom-team , 

Please cover this case as well
SKU 1: 100$
SKU 2: 200$
Rule: Fix amount 150$ for whole cart(CartFixed)
i add to cart:
SKU 1 - qty 1
SKU 2 - qty 1
then checkout:
SKU 1 - address 1
SKU 2 - address 2

### Expectation:
after checkout:
Sale Order 1: discount amount 100$
Sale Order 2: discount amount: 50$

Thanks

**Comment by bubasuma** [comment]
Internal Magento team is working on this issue right now.

**Comment by magento-engcom-team** [comment]
Hi @hoangnm89, @bubasuma, @maheshWebkul721. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.4-develop` branch(es).
Related commit(s):
 - [b6f84336ac9f0059b76146206a8483d5e4e7c81e](https://github.com/magento/magento2/commit/b6f84336ac9f0059b76146206a8483d5e4e7c81e)

The fix will be available with the upcoming `2.4.0` release.
